doc/tutorials/enable_ivshmem.rst

```diff
@@ -0,0 +1,151 @@
+
+	-  ``VM IDs``   - Specify the VM IDs to use the same shared memory
+	   communication and separate it with ``:``. For example, the
+	   communication between VM0 and VM2, it can be written as ``0:2``
+
+- Build the XML configuration, refer to :ref:`Build ACRN from Source <getting-started-building>`
+
+Inter-VM Communication Example
+******************************
+
+**The following example uses inter-vm communication(dm-land) between two
+Linux-based post-launched VMs (VM1 and VM2)**
+
+.. note:: An ``ivshmem`` Windows driver exists and can be found `here <https://github.com/virtio-win/kvm-guest-drivers-windows/tree/master/ivshmem>`_
+
+1. Add a new virtual PCI device for both VMs: the device type is
+   ``ivshmem``, shared memory name is ``sos:/test``, and shared memory size is
+   2097152(2M) bytes. Both VMs must have the same shared memory name and size:
+
+   - VM1 Launch Script Sample
+
+     .. code-block:: none
+        :emphasize-lines: 7
+
+        acrn-dm -A -m $mem_size -s 0:0,hostbridge \
+         -s 2,pci-gvt -G "$2" \
+         -s 5,virtio-console,@stdio:stdio_port \
+         -s 6,virtio-hyper_dmabuf \
+         -s 3,virtio-blk,/home/acrn/uos1.img \
+         -s 4,virtio-net,tap0 \
+         -s 6,ivshmem,sos:/test,2097152\
+         -s 7,virtio-rnd \
+         --ovmf /usr/share/acrn/bios/OVMF.fd \
+         $vm_name
+
+
+   - VM2 Launch Script Sample
+
+     .. code-block:: none
+        :emphasize-lines: 5
+
+        acrn-dm -A -m $mem_size -s 0:0,hostbridge \
+         -s 2,pci-gvt -G "$2" \
+         -s 3,virtio-blk,/home/acrn/uos2.img \
+         -s 4,virtio-net,tap0 \
+         -s 5,ivshmem,sos:/test,2097152\
+         --ovmf /usr/share/acrn/bios/OVMF.fd \
+         $vm_name
+
+2. Boot two VMs and use ``lspci | grep "shared memory"`` to verify that the virtual device is ready for each VM.
+
+   -  For VM1, it shows ``00:06.0 RAM memory: Red Hat, Inc. Inter-VM shared memory (rev 01)``
+   -  For VM2, it shows ``00:05.0 RAM memory: Red Hat, Inc. Inter-VM shared memory (rev 01)``
+
+3. Use these commands to probe the device::
+
+     $ sudo modprobe uio
+     $ sudo modprobe uio_pci_generic
+     $ sudo echo "1af4 1110" > /sys/bus/pci/drivers/uio_pci_generic/new_id
+
+4. Finally, a user application can get the shared memory base address from
+   the ``ivshmem`` device BAR resource
+   (``/sys/class/uio/uioX/device/resource2``) and the shared memory size from
+   the ``ivshmem`` device config resource
+   (``/sys/class/uio/uioX/device/config``).
+
+   The ``X`` in ``uioX`` above, is a number that can be retrieved using the
+   ``ls`` command:
+
+   - For VM1 use ``ls -lh /sys/bus/pci/devices/0000:00:06.0/uio``
+   - For VM2 use ``ls -lh /sys/bus/pci/devices/0000:00:05.0/uio``
+
+**The following example uses inter-vm communication(hv-land) between two
+Linux-based VMs (VM0 is a pre-launched VM and VM2 is a post-launched VM)**
+
+1. Configure shared memory for the communication between VM0 and VM2 in hybrid_rt
```

Comment:
In this example, you're editing xml files where in the other example you just did command line stuff.  Are these two ways of doing the same thing?  If so, we should write that here.

Reply:
If user wants to use hv-land, user needs to edit XML file. For dm-land, user only do command line stuff.
So, this sample is hv-land and  user needs to edit hybrid_rt scenario and whl-ipc-i5 board xml file